version_control_pattern.md

```diff
@@ -30,17 +30,19 @@ git commit
 
 # This will take you into your default text editor
 # Write a descriptive commit message
-
-# If you have not configured your text editor, you may get stuck in Vim
-# exit using the following command:
-:q!
-
-# Then configure your default text editor for Git
+```
+>**Note:**
+>If you have not configured your text editor, you may get stuck in Vim. You can exit using the following command: `:q!`. Then configure your default text editor for Git
+```sh
 git config --global core.editor <my-favourite-text-editor>
-
-# And try again
+ # Then try again
 git commit
 ```
+>You can also commit and add a message to that commit at the same time
+```sh
+git commit -m "A useful message explaining my changes"
```

Comment:
I'd advise against doing this.  It saves almost no time or effort and denies the possibility to double check what files are being committed (because when the commit message is entered using the text editor which pops open, you can see a list of changes).  You *may* also get some feedback which helps remind you to format your commit message in the 'correct' way.  I.e. Less than 50 chars, starting with a capital letter.

Reply:
For reference: https://robots.thoughtbot.com/5-useful-tips-for-a-better-commit-message

Reply:
Fair point - will remove this bit before next commit.